test/simulation_and_solving/jacobian_construction.jl

```diff
@@ -67,4 +67,107 @@ let
                     p[k3]*u[B] p[k3]*u[A] -p[k4]-3 * p[k5] * u[C]^2 / 2]
         @test test_jac ≈ real_jac
     end
-end
\ No newline at end of file
+end
+
+# Checks that the Jacobians (dense and sparse) are identical for different system types.
```

Comment:
Do you ever test against a custom hand-coded Jacobian we know is right? That would be good to have in at least one case.

Reply:
Yes we do (right now that's basically the only Jacobian tests we do actually): https://github.com/SciML/Catalyst.jl/blob/master/test/simulation_and_solving/jacobian_construction.jl

However, we do not do this for `sparse` though.

Reply:
We should check sparse against the exact Jacobian, but I don't think we need an "exact" Jacobian that returns a sparse matrix (i.e. checking the Catalyst-generated sparse vs. a hand dense is fine).